### PR TITLE
Added Tumblr provider

### PR DESCRIPTION
**Pull request by neemzy**
Here is an implementation of a Tumblr provider. Share count isn't provided as the info isn't available without an API key.

### CODE REVIEW

**Comment by dunglas** [comment]
It would be nice to have optional count retrieving support even if it requires an API key. The key can be injected in the constructor of the provider.

**Comment by neemzy** [comment]
You're right, I'll work on it.

**Comment by dunglas** [comment]
@neemzy I plan to release the 1.0 version in the next few days, do you think this PR can be finished soon or it will wait for 1.1?

**Comment by neemzy** [comment]
What's your deadline? I can deliver in the upcoming week.

**Comment by neemzy** [comment]
After taking a closer look, it appears that the only count exposed by the Tumblr API is the reblog count for a Tumblr post - one can share any URL, but total share count for this URL cannot be retrieved.

**Comment by dunglas** [comment]
Ok so I merge it as is?

File: src/SocialShare/Provider/Tumblr.php [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace SocialShare\Provider;
+
+class Tumblr implements ProviderInterface
+{
+    const NAME = 'tumblr';
+    const SHARE_URL = 'http://www.tumblr.com/share/link?%s';
```
Comment (dunglas) [inline src/SocialShare/Provider/Tumblr.php]: Would be nice to use https by default.

File: src/SocialShare/Provider/Tumblr.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        return sprintf(self::SHARE_URL, http_build_query($options, null, '&'));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function getShares($url)
+    {
+        // This information cannot be retrieved from Tumblr API
+        return 0;
```
Comment (dunglas) [inline src/SocialShare/Provider/Tumblr.php]: Maybe than trowing an exception is better than silently returning 0?

File: README.md [hunk 3]
```diff
@@ -38,6 +39,8 @@ These services can be shut down without warning.
 
 Google returns only an estimate when there is more than 1000 shares (something like >10K). PHP Social Share converts this estimate to an integer.
 
+Tumblr share count cannot be retrieved without an API key, and is therefore unavailable through this library.
```
Comment (dunglas) [inline README.md]: Should be updated to reflect that the count number cannot be retrieved at all.

**Comment by neemzy** [comment]
Good points! I fixed them. I reckon merging as-is is our best option right know.

**Comment by dunglas** [comment]
Ok I'll merge it when specs will be updated.

**Comment by dunglas** [comment]
@neemzy can you update the specs and squash the commit? Then I'll release 1.0.

**Comment by neemzy** [comment]
Yep, sorry, didn't find time until right now to fix this. There you go!

**Comment by dunglas** [comment]
Thank you!

**Comment by neemzy** [comment]
Cheers!

**Pull request merged**